gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java

```diff
@@ -0,0 +1,526 @@
+     * Adds a DNS record of type A. The last parameter is ttl and is not required. If ttl is not
+     * provided, a default value of 0 will be used.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      int ttl = 0;
+      if (args.length > 3) {
+        ttl = Integer.parseInt(args[3]);
+      }
+      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+          .records(ImmutableList.of(ip))
+          .ttl(ttl, TimeUnit.SECONDS)
+          .build();
+      ChangeRequest changeRequest = ChangeRequest.builder()
+          .add(record)
+          .build();
+      changeRequest = dns.applyChangeRequest(zoneName, changeRequest);
+      System.out.printf("The request for adding A record %s for zone %s was successfully " +
+          "submitted and assigned ID %s.%n", recordName, zoneName, changeRequest.id());
+      System.out.print("Waiting for addition to happen...");
+      while (changeRequest.status().equals(ChangeRequest.Status.PENDING)) {
+        System.out.print(".");
+        try {
+          Thread.sleep(500);
+        } catch (InterruptedException e) {
+          System.err.println("Thread was interrupted while waiting.");
+        }
+        changeRequest = dns.getChangeRequest(zoneName, changeRequest.id());
+      }
+      System.out.printf("The addition has been completed.%n");
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> <record_name> <ip_address> [<ttl>]";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      if (args.length == 4) {
+        // to check that it can be parsed
+        Integer.parseInt(args[3]);
+        return true;
+      } else {
+        return args.length == 3;
+      }
+    }
+  }
+
+  private static class ListDnsRecordsAction implements DnsAction {
+
+    /**
+     * Lists all the DNS records in the given zone.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Iterator<DnsRecord> iterator = dns.listDnsRecords(zoneName).iterateAll();
+      if (iterator.hasNext()) {
+        System.out.printf("DNS records for zone %s:%n", zoneName);
+        while (iterator.hasNext()) {
+          DnsRecord record = iterator.next();
+          System.out.printf("%nRecord name: %s%nTTL: %s%nRecords: %s%n", record.name(),
+              record.ttl(), Joiner.on(", ").join(record.records()));
+        }
+      } else {
+        System.out.printf("Zone %s has no DNS records.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> records";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 2;
+    }
+  }
+
+  private static class ListChangesAction implements DnsAction {
+
+    /**
+     * Lists all the changes for a given zone. Optionally, an order, "descending" or "ascending" can
```

Comment:
Nit, add a "," after "ascending".  To avoid too many commas, you could also put "descending" or "ascending" in parens (and then there's no need to add a comma after ascending)